api/v1alpha4/tags.go

```diff
@@ -30,7 +30,7 @@ type Tags map[string]string
 
 // Equals returns true if the tags are equal.
 func (t Tags) Equals(other Tags) bool {
-	return reflect.DeepEqual(t, other)
+	return cmp.Equal(t, other)
```

Comment:
Even thid Equals function is not used, we can remove it as well.

Reply:
will do it now. But still one test is failing though it is non-required one. 
Is that okay to fail @Ankitasw ??

Reply:
You mean apidiff-main? We can ignore it for these changes.

Reply:
Once these changes are done, please squash you commits as well.

Reply:
and on the other hand one Equals function is also there in api/v1beta1/tags.go version.
should i delete that as well @Ankitasw ?

Reply:
Actually, on thinking more we should not delete this function. It's defined on our API types and we may have people using this (as a package) so we will need to deprecate the function instead. wdyt @Ankitasw ?

Reply:
sounds good to me @richardcase

Reply:
@VibhorChinda sorry for changing the decision :) you can leave all Equals func as is, just that in the comments, you can mention that this func is deprecated.

Reply:
Its just the `Tags.Equals` functional we shouldn't delete (on further thinking). It can still use cmp.Equals internally though like you originally had. Sorry for initially saying we could delete this function....i should've thought it through some more.

Reply:
No worries. Will do so :)